tests/test_utils.py

```diff
@@ -51,6 +54,13 @@ def _setup_for_generation(instance, test_name):
     )
 
 
+def _bom_entry_sort_key(x: BomEntry) -> str:
```

Comment:
I think we should wrap the bom writing code and use it in the tests. Should avoid the duplication between the test and generate_bom.py file